xwiki-platform-core/xwiki-platform-mail/xwiki-platform-mail-send/xwiki-platform-mail-send-default/src/main/java/org/xwiki/mail/script/ScriptMimeMessage.java

```diff
@@ -79,16 +79,29 @@
         this.execution = execution;
         this.componentManager = componentManager;
     }
+    
+    /**
+     * Add some content to the mail to be sent. Can be called several times to add different content type to the mail.
+     *
+     * @return the Mime Body Part object that was added. Returning it allows script to make modifications to that body
+     *         part after it's been set (get/set some headers, etc)
+     * @param bodyPart a {@link BodyPart} object to include in the mail
+     */
+    public BodyPart addPart(BodyPart bodyPart)
+    {
+        return addPart(null, bodyPart);
+    }
 
     /**
      * Add some content to the mail to be sent. Can be called several times to add different content type to the mail.
      *
      * @return the Mime Body Part object that was added. Returning it allows script to make modifications to that body
      *         part after it's been set (get/set some headers, etc)
      * @param mimeType the mime type of the content parameter
-     * @param content the content to include in the mail. The type depends on the mimetype. For example for a mime
-     *        type of {@code text/plain} or {@code text/html}, the content should be a String, for a mime type of
-     *        {@code xwiki/template} it should be a {@link org.xwiki.model.reference.DocumentReference}, etc.
+     * @param content the content to include in the mail. The type depends on the mimetype. For example for a mime type
+     *            of {@code text/plain} or {@code text/html}, the content should be a String, for a mime type of
+     *            {@code xwiki/template} it should be a {@link org.xwiki.model.reference.DocumentReference}, etc. Also
+     *            accepts a {@link BodyPart} object, in which case, the mime type is not used.
```

Comment:
This shouldn't be part of your PR since this is something I committed!

Reply:
I just formatted it using ctrl shift f, that's why its showing as added by me

Reply:
Indeed. Why did you do that? :) Best is to follow the same strategy as what was there before ;) Would be good if you could undo that. Thanks!

Reply:
Sure!